src/projects/TodoList/index.js

```diff
@@ -10,21 +12,43 @@ class TodoList extends React.Component {
     this.pressEnter = this.pressEnter.bind(this)
     this.changeCompleted = this.changeCompleted.bind(this)
     this.confirmEditTask = this.confirmEditTask.bind(this)
+    this.toggleCheckBox = this.toggleCheckBox.bind(this)
+    this.addCategory = this.addCategory.bind(this)
     this.state = {
       todos: [
-        { task: 'Take out the bins', completed: false, isEditing: false, editingTask: '' },
-        { task: 'Put a wash load on', completed: true, isEditing: false, editingTask: '' },
-        { task: 'Learn React', completed: false, isEditing: false, editingTask: '' }
+        { task: 'Take out the bins', completed: false, isEditing: false, editingTask: '', categories: [1,2,3,4] },
+        { task: 'Put a wash load on', completed: true, isEditing: false, editingTask: '', categories: [1,4] },
+        { task: 'Learn React', completed: false, isEditing: false, editingTask: '', categories: [3] }
       ],
-      newItem: ''
+      newItem: '',
+      categories: [
+        { id: 1, category: "Urgent", colour: "#dc3545", text: "white", checked: false },
+        { id: 2, category: "Housework", colour: "#007bff", text: "white", checked: false },
+        { id: 3, category: "Shopping", colour: "#28a745", text: "white", checked: false },
+        { id: 4, category: "Birthday", colour: "#ffc107", text: "black", checked: false }
+      ],
+      newCategory: '',
+      newCategoryRed: '0',
+      newCategoryGreen: '0',
+      newCategoryBlue: '0',
+      newCategoryTextColour: 'white',
+      newCategoryBeingCreated: false
     }
   }
   addItem() {
     const newTodos = this.state.todos
-    if (this.state.newItem !== '') {
-      const newTodo = { task: this.state.newItem, completed: false }
+    if (this.state.newItem.trim() !== '') {
+      const checkedCategories = this.state.categories.filter(todo => todo.checked === true)
```

Comment:
This filter function can be slimmed down a tiny tiny bit - You know how in your filter you're saying "if todo is checked then return true"; think about that line for a second. The thing that matters from a filter function is whether you return true or false.
```js
this.state.categories.filter(todo => todo.checked === true)
```
can be written as
```js
this.state.categories.filter(todo => todo.checked)
```
Because todo.checked is true or false so if you just return whether todo.checked is true or false you achieve the same thing as saying `todo.checked === true`

Reply:
yep, good point. done